machine-learning/tutorials/TransferLearningTF/Program.cs

```diff
@@ -0,0 +1,206 @@
+            // </SnippetTrainModel>
+
+            // Process the training data through the model
+            // This is an optional step, but it's useful for debugging issues
+            // <SnippetTransformData>
+            var predictions = model.Transform(data);
+            // </SnippetTransformData>
+
+            // Create enumerables for both the ImageData and ImagePrediction DataViews 
+            // for displaying results
+            // <SnippetEnumerateDataViews>
+            var imageData = mlContext.CreateEnumerable<ImageData>(data, false, true);
+            var imagePredictionData = mlContext.CreateEnumerable<ImagePrediction>(predictions, false, true);
+            // </SnippetEnumerateDataViews>
+
+            // Read the tags.tsv file and add the filepath to the image file name 
+            // before loading into ImageData 
+            // <SnippetCallPairAndDisplayResults1>
+            PairAndDisplayResults(imageData, imagePredictionData);
+            // </SnippetCallPairAndDisplayResults1>
+
+            // Get some performance metrics on the model using training data
+            Console.WriteLine("=============== Classification metrics ===============");
+
+            // <SnippetEvaluate>           
+            var regressionContext = new MulticlassClassificationCatalog(mlContext);
+            var metrics = regressionContext.Evaluate(predictions, label: LabelTokey, predictedLabel: DefaultColumnNames.PredictedLabel);
+            // </SnippetEvaluate>
+
+            //<SnippetDisplayMetrics>
+            Console.WriteLine($"LogLoss is: {metrics.LogLoss}");
+            Console.WriteLine($"PerClassLogLoss is: {String.Join(" , ", metrics.PerClassLogLoss.Select(c => c.ToString()))}");
+            //</SnippetDisplayMetrics>
+
+            // Save the model to assets/outputs
+            Console.WriteLine("=============== Save model to local file ===============");
+
+            // <SnippetSaveModel>
+            using (var fileStream = new FileStream(outputModelLocation, FileMode.Create))
+                mlContext.Model.Save(model, fileStream);
+            // </SnippetSaveModel>
+
+            Console.WriteLine($"Model saved: {outputModelLocation}");
+        }
+
+        public static void ClassifyImages(MLContext mlContext, string dataLocation, string imagesFolder, string outputModelLocation)
+        {
+            Console.WriteLine($"=============== Loading model ===============");
+            Console.WriteLine($"Model loaded: {outputModelLocation}");
+
+            // Load the model
+            // <SnippetLoadModel>
+            ITransformer loadedModel;
+            using (var fileStream = new FileStream(outputModelLocation, FileMode.Open))
+                loadedModel = mlContext.Model.Load(fileStream);
+            // </SnippetLoadModel>
+
+            // Read the tags.tsv file and add the filepath to the image file name 
+            // before loading into ImageData 
+            // <SnippetReadFromCSV> 
```

Comment:
Since method name changed from `ReadFromCsv` to `ReadFromTsv`, change snippet name to SnippetReadFromTSV to avoid confusion.

Reply:
Good catch. It's changed in the Sample.

Reply:
Yup, good catch. Already updated in tutorial

Reply:
Done!